server/src/main/java/io/crate/execution/ddl/tables/DropColumnTask.java

```diff
@@ -84,17 +88,33 @@ public ClusterState execute(ClusterState currentState, DropColumnRequest request
                                         return dc.ref();
                                     });
 
-        Metadata.Builder metadataBuilder = Metadata.builder(currentState.metadata());
-        Map<String, Object> mapping = createMappingForDroppedCols(currentTable, refsToDrop);
+        // Prepare columns tree, it will be used twice:
+        // 1. For creating mapping for dropped columns.
+        // 2. For creating "path-only, no properties" mapping to remove.
+        List<Reference> references = new ArrayList<>(refsToDrop);
+        for (var colToDrop : refsToDrop) {
+            ColumnIdent parent = colToDrop.column();
+            while ((parent = parent.getParent()) != null) {
+                if (!Symbols.containsColumn(refsToDrop, parent)
+                    && !Symbols.containsColumn(references, parent)) {
+                    references.add(currentTable.getReference(parent));
+                }
+            }
+        }
+        HashMap<ColumnIdent, List<Reference>> tree = buildTree(references);
+
+        Map<String, Object> mapping = createMappingForDroppedCols(currentTable, tree);
+        Map<String, Object> mappingToRemove = createMappingToRemove(tree, null);
```

Comment:
This is taken out from `MappingUtil.createMappingForDroppedCols` as we traverse the tree twice.

Reply:
Because we operate only with request columns and at the end merge it into existing mapping. We don't re-create mapping for existing columns.

If we replace keys in one traversal, we would just add prefixed entries to the existing mapping -> but we still need to get rid of non-prefixed entries.

What's important is that Map<String, Object> mapping and Map<String, Object> mappingToRemove have different keys - prefixed and non-prefixed correspondingly.

We can probably create a traversing method which creates Tuple of maps but it looks a bit overkill for DDL operation. Also can take all request columns and remove "merge" logic and just replace mapping - but I feel like awe shouldn't regenerate existing entries, it's also not nice.